deployment_logger.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// error messages
+var (
+	ErrLoggerNotInitialized = errors.New("logger not initialized")
+)
+
+type FileLogger struct {
+	logFileName string
+	logFile     io.WriteCloser
+}
+
+// NewFileLogger creates instance of file logger; it is initialized
+// just before logging is started
+func NewFileLogger(name string) *FileLogger {
+	// open log file
+	logFile, err := os.OpenFile(name, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
```

Comment:
One thing I'm not sure of, but maybe it makes more sense to open & close the file in `Write()` rather than doing it here and dragging `io.WriteCloser` around.